### PR TITLE
Use x-scope annotations when resolving refs during request/response validation

### PR DESCRIPTION
**Pull request by analogue**
Fix for #82

Unit tests have been added.

### CODE REVIEW

**Comment by sjaensch** [comment]
This is awesome, thanks for doing it! I'll try to work on unit tests in the evening if I get a chance.

File: bravado_core/swagger20_validator.py [hunk 1]
```diff
@@ -87,6 +88,45 @@ def enum_validator(swagger_spec, validator, enums, instance, schema):
...
+def ref_validator(validator, ref, instance, schema):
+    """When validating a request or response that contain $refs, jsonschema's
+     RefResolver only contains scope (RefResolver._scopes_stack) for the
+     request_spec or response_spec that it is fed. This scope doesn't contain
+     the full scope built when ingesting the spec from its root
+     (#/ in swagger.json). So, we need to modify the behavior of ref
+     validation to use the `x-scope` annotations that were created during spec
+     ingestion (see post_process_spec in spec.py).
+
+    :param swagger_spec: needed for access to deref()
```
Comment (sjaensch) [inline bravado_core/swagger20_validator.py]: You're not passing in a spec.

Comment (analogue) [inline bravado_core/swagger20_validator.py]: nice catch - removed

**Comment by analogue** [comment]
Superceded by #84

**Pull request closed without merging**